stratum/hal/lib/barefoot/bf_sde_wrapper.cc

```diff
@@ -1324,6 +1327,51 @@ ::util::Status BfSdeWrapper::SetDeflectOnDropDestination(int device, int port,
 
 // BFRT
 
+::util::Status BfSdeWrapper::InitializeSde(const std::string& sde_install_path,
+                                           const std::string& sde_config_file,
+                                           bool run_in_background) {
+  CHECK_RETURN_IF_FALSE(sde_install_path != "")
+      << "sde_install_path is required";
+  CHECK_RETURN_IF_FALSE(sde_config_file != "") << "sde_config_file is required";
+
+  // Parse bf_switchd arguments.
+  auto switchd_main_ctx = absl::make_unique<bf_switchd_context_t>();
+  switchd_main_ctx->install_dir = strdup(sde_install_path.c_str());
+  switchd_main_ctx->conf_file = strdup(sde_config_file.c_str());
+  switchd_main_ctx->skip_p4 = true;
+  if (run_in_background) {
+    switchd_main_ctx->running_in_background = true;
+  } else {
+    switchd_main_ctx->shell_set_ucli = true;
+  }
+
+  // Determine if kernel mode packet driver is loaded.
+  std::string bf_sysfs_fname;
+  {
+    char buf[128] = {};
+    CHECK_RETURN_IF_FALSE(
+        switch_pci_sysfs_str_get(buf, sizeof(buf) - sizeof("/dev_add")) == 0);
```

Comment:
```suggestion
        switch_pci_sysfs_str_get(buf, sizeof(buf)) == 0);
```

The extra bytes are not needed when using C++ strings later.